sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogSuite.scala

```diff
@@ -181,41 +182,47 @@ class HiveExternalCatalogSuite extends ExternalCatalogSuite {
       "INSERT overwrite directory \"fs://localhost/tmp\" select 1 as a"))
   }
 
-  test("SPARK-31061: alterTable should be able to change table provider") {
+  test("SPARK-31061: alterTable should be able to change table provider/hive") {
     val catalog = newBasicCatalog()
-    val parquetTable = CatalogTable(
-      identifier = TableIdentifier("parq_tbl", Some("db1")),
-      tableType = CatalogTableType.MANAGED,
-      storage = storageFormat.copy(locationUri = Some(new URI("file:/some/path"))),
-      schema = new StructType().add("col1", "int").add("col2", "string"),
-      provider = Some("parquet"))
-    catalog.createTable(parquetTable, ignoreIfExists = false)
```

Comment:
@brkyvz  @cloud-fan [https://github.com/apache/spark/pull/27822](https://github.com/apache/spark/pull/27822) 
I find `storage = storageFormat.copy(locationUri = Some(new URI("file:/some/path"))),` maybe useless, and will cause the path incompatible with path properties. So shall we remove it and merge the two test cases to one ?

Reply:
why does it cause path incompatibility? `storageFormat` doesn't have a path property.

Reply:
Because UT failed and tell us：`path in location Some(file:/tmp/spark-cc7f24a5-e626-4d91-afc6-4f2702482980/parq_tbl) not equal to table prop path Some(file:/some/path)`, could I send another PR to polish this.

Reply:
Yes please. This test looks like a valid one and should not fail.

Reply:
Yes, it is a valid case, and it seems there are something to improve, [https://github.com/apache/spark/pull/28980](https://github.com/apache/spark/pull/28980)